internal/component/common/sink.go

```diff
@@ -132,3 +137,65 @@ func (c *commonSink) JSONPathSelector(data []byte, path string) ([]byte, error)
 	}
 	return raw, nil
 }
+
+// Compress compresses the given filepaths using the specified compression type and password.
+// It's useful for compressing files before sending them to the sink.
+func (c *commonSink) Compress(compressionType, compressionPassword string, filepaths []string) ([]string, error) {
+	if compressionType == "" || len(filepaths) == 0 {
+		return filepaths, nil
+	}
+
+	var archivedPaths []string
+	switch compressionType {
+	case "gz", "gzip":
+		for _, filePath := range filepaths {
+			fileName := fmt.Sprintf("%s.%s", filepath.Base(filePath), compressionType)
+			archivedPath := filepath.Join(filepath.Dir(filePath), fileName)
+			archivedPaths = append(archivedPaths, archivedPath)
+
+			f, err := os.OpenFile(archivedPath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+			if err != nil {
+				return nil, errors.WithStack(err)
+			}
+			defer f.Close()
+
+			archiver := archive.NewFileArchiver(c.Logger(), archive.WithExtension(compressionType))
+			if err := archiver.Archive([]string{filePath}, f); err != nil {
+				return nil, errors.WithStack(err)
+			}
+		}
+	case "zip", "tar.gz":
+		// for zip & tar.gz file, the whole file is archived into a single archive file
+		// single archive file is created in the same directory as the nearest common parent directory of the filepaths
+		// eg. if filepaths are /tmp/a/b/c.txt and /tmp/a/d/e.txt, the archive file will be created as /tmp/a/archive.zip
+
+		// get the nearest common parent directory of the filepaths
+		dir := filepath.Dir(filepaths[0])
+		for _, filePath := range filepaths[1:] {
+			parentDir := filepath.Dir(filePath)
+			i := 0
+			for ; i < len(strings.Split(dir, "/")) && strings.Split(dir, "/")[i] == strings.Split(parentDir, "/")[i]; i++ {
+			}
+			dir = strings.Join(strings.Split(dir, "/")[:i], "/")
+		}
+
+		fileName := fmt.Sprintf("archive.%s", compressionType)
```

Comment:
if there is only one destination file (without templating), should the naming use the original filename instead of `archive.<extension>`?